app/models/product_export.rb

```
@@ -0,0 +1,157 @@
+class ProductExport < ApplicationRecord
+  has_one_attached :export_file
+
+  def export(products)
+    Axlsx::Package.new do |p|
+      book = p.workbook
+
+      add_product_info_worksheet(book, products)
+
+      add_test_results_worksheet(book, products)
+
+      add_risk_assessments_worksheet(book, products)
+
+      add_corrective_actions_worksheet(book, products)
+
+      p.serialize(Rails.root.join("product_export_#{id}.xlsx"))
+
+      export_file.attach(io: File.open(Rails.root.join("product_export_#{id}.xlsx")), filename: "products_export.xlsx")
+
+    rescue StandardError => e
+      Sentry.capture_exception(e)
```

I think this should be in the parent job class as we might want to run this in the console for debugging for example.

Query: How do we want to handle failure? This will send us an alert but the user would not receive the email they're expecting. Should we re-raise the exception  after reporting to Sentry so that it enters the Sidekiq retry queue? Or are we going to requeue them manually once any issue is fixed?

---

Hmmm re-raising them sounds like the way to go, good idea.